internal/utils/cli/error.go

```diff
@@ -0,0 +1,138 @@
+package cli
+
+import (
+	"fmt"
+)
+
+// ErrUsageHider may hide the command's usage when an error occurs
+type ErrUsageHider interface {
+	HideUsage() bool
+}
+
+// ErrSuggester provides a list of suggestions that will display to the user when an error occurs
+type ErrSuggester interface {
+	Suggestions() []interface{}
+}
+
+// ErrLinkReferrer provides a list of reference links that will display to the user when an error occurs
+type ErrLinkReferrer interface {
+	ReferenceLinks() []interface{}
+}
+
+// NewErr returns a new CLI error
+func NewErr(cause error, details ...ErrDetail) error {
+	var d ErrDetails
+	for _, detail := range details {
+		detail.Apply(&d)
+	}
+
+	var hideUsage bool
+	if d.HideUsage != nil {
+		hideUsage = *d.HideUsage
+	}
+
+	return cliErr{
+		cause:          cause,
+		hideUsage:      hideUsage,
+		suggestions:    d.Suggestions,
+		referenceLinks: d.ReferenceLinks,
+	}
+}
+
+// WrapErr wraps the provided error with the provided message
+// The extra details are applied on top of any existing, wrapped details
+// Note: it is assumed the message contains a single `fmt` verb to include the cause/error
```

Comment:
> // Note: it is assumed the message contains a single `fmt` verb to include the cause/error

q: can you clarify this a bit more? I'm not really sure what this means

Reply:
EDIT: Ah i see now. Maybe we can rename the `msg` argument to `msgFormat`?

This is what `fmt.Sprintf` looks like:

<img width="739" alt="image" src="https://user-images.githubusercontent.com/15252988/153081987-bef6e9cc-75cb-448a-b83e-5d0082cf7d70.png">

Reply:
to be fair, I don't think many people would like the name "format" for what this is either... pattern or template maybe, but those are also already first-class things in the language.  i'll change the arg name here, but my hope is people look to the code for what this is doing over the arg name